initialize/config.go

```diff
@@ -39,17 +38,51 @@ type CloudConfigUnit interface {
 	Units() []system.Unit
 }
 
+func ApplyHostname(hostname string) error {
+	if err := system.SetHostname(hostname); err != nil {
+		return fmt.Errorf("error setting hostname: %w", err)
+	}
+	log.Printf("Set hostname to %s", hostname)
+	return nil
+}
+
+func ApplyCoreUserSSHKeys(keys []string, env *Environment) error {
+	if len(keys) == 0 {
+		return nil
+	}
+	err := system.AuthorizeSSHKeys("core", env.SSHKeyName(), keys)
+	if err == nil {
+		log.Printf("Authorized SSH keys for core user")
+	} else {
+		return err
+	}
+
+	return nil
```

Comment:
This looks weird, how about:
```suggestion
	if err != nil {
		return err
	}
	log.Printf("Authorized SSH keys for core user")
	return nil
```